From 9c32ff2e3bc614e6cbf2c05078470ef6204811a7 Mon Sep 17 00:00:00 2001
From: Linu Cherian <lcherian@marvell.com>
Date: Mon, 30 Mar 2020 12:44:27 +0530
Subject: [PATCH 512/767] coresight: Increase register poll timeout

commit da10b15d511fe3ada6e889d5b634bfab09b5d882 from
git@git.assembla.com:cavium/WindRiver.linux.git

On OcteonTx2 implementation, ETR flush does takes few
milliseconds to complete. Hence increasing register
poll timeout to 5ms.

Change-Id: I5de253dae781d9074d09f168d31d6a965ec16349
Signed-off-by: Linu Cherian <lcherian@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/26058
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/hwtracing/coresight/coresight-priv.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/hwtracing/coresight/coresight-priv.h b/drivers/hwtracing/coresight/coresight-priv.h
index 31d801707ccd..f54abf5399ff 100644
--- a/drivers/hwtracing/coresight/coresight-priv.h
+++ b/drivers/hwtracing/coresight/coresight-priv.h
@@ -33,7 +33,10 @@
  */
 #define CORESIGHT_CLAIM_SELF_HOSTED	BIT(1)
 
-#define TIMEOUT_US		100
+/* Timeout is in ms to accommodate larger ETR flush time
+ * taken by OcteonTx2 implementation
+ */
+#define TIMEOUT_US		5000
 #define BMVAL(val, lsb, msb)	((val & GENMASK(msb, lsb)) >> lsb)
 
 #define ETM_MODE_EXCL_KERN	BIT(30)
-- 
2.31.1

